uflash.py

```diff
@@ -72,8 +75,53 @@ def find_microbit():
 
     Returns None if no micro:bit is found.
     """
-    # TODO: FUNKYHAT'S WORK HERE!!!!!
-    pass
+    if os.name == 'posix':
+        mount_output = check_output('mount').splitlines()
+        # (almost) list the mounted volumes.
+        mounted_volumes = [x.split()[2] for x in mount_output]
+
+        for volume in mounted_volumes:
+            if volume.endswith('MICROBIT'):
+                return volume
+
+    elif os.name == 'nt':
+        kernel32 = ctypes.windll.kernel32
+
+        vol_name_buf = ctypes.create_unicode_buffer(1024)
+        fs_name_buf = ctypes.create_unicode_buffer(1024)
+
+        serial_number = max_component_length = file_system_flags = None
+
+        def get_volume_name(volume):
+            """
+            Code from http://stackoverflow.com/a/12056414
```

Comment:
As well as crediting the source, can we add a description of what it does? E.g. "Returns the volume name of the referenced drive. "